### ISSUE TITLE
My new problem after updating Ubuntu to 17.04

### ISSUE THREAD

**Issue by Lourens-Rich** [go-gitea/gitea#1496]
I decided to update my instance and Golang environment. Used this commit: https://github.com/go-gitea/gitea/commit/237270ef50ba00b3007ea10c2302bc240188e092

```
./gitea web
2017/04/16 00:15:10 [T] Custom path: /home/git/go/src/code.gitea.io/gitea/custom
2017/04/16 00:15:10 [T] Log path: /home/git/go/src/github.com/gogits/gogs/log
2017/04/16 00:15:10 [I] Gitea v1.1.0+dev
{0 0} &{version migrations.Version [id version] map[id:[0xc4202de600] version:[0xc4202de6c0]] [0xc4202de600 0xc4202de6c0] map[] [id] id map[]    <nil>  } <nil>
```
```
cat ~/go/src/github.com/gogits/gogs/log/gitea.log 
2017/04/16 00:16:59 [I] Log Mode: File(Info)
2017/04/16 00:16:59 [I] XORM Log Mode: File(Info)
2017/04/16 00:16:59 [I] Cache Service Enabled
2017/04/16 00:16:59 [I] Session Service Enabled
2017/04/16 00:16:59 [I] Migration: change mirror interval from hours to time.Duration
2017/04/16 00:16:59 [...itea/routers/init.go:54 GlobalInit()] [E] Failed to initialize ORM engine: migrate: do migrate: update mirror interval failed: Error 1264: Out of range value for column 'interval' at row 1
```
```
cat ~/go/src/github.com/gogits/gogs/log/xorm.log
2017/04/16 00:16:59 [I] PING DATABASE mysql
2017/04/16 00:16:59 [I] [SQL] SELECT `TABLE_NAME` from `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_SCHEMA`=? and `TABLE_NAME`=? [gogs version]
2017/04/16 00:16:59 [I][sql]SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = ? AND `TABLE_NAME` = ? AND `COLUMN_NAME` = ?[gogs version id]
2017/04/16 00:16:59 [I][sql]SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = ? AND `TABLE_NAME` = ? AND `COLUMN_NAME` = ?[gogs version version]
2017/04/16 00:16:59 [I] [SQL] SELECT `id`, `version` FROM `version` WHERE `id`=? LIMIT 1 [1]
2017/04/16 00:16:59 [I] [SQL] BEGIN TRANSACTION
2017/04/16 00:16:59 [I] [SQL] SELECT * FROM `mirror`
2017/04/16 00:16:59 [I] [SQL] UPDATE `mirror` SET `interval` = ? WHERE `id`=? [8h0m0s 1]
2017/04/16 00:16:59 [I] [SQL] ROLL BACK
```

Sorry, could you help me?

**Comment by sapk** [comment]
Just to be sure, your database is mysql ? 
Can you see if the format of the column interval (in table mirror) is a string or a integer at this stage of migration ?
Are you migrating from gogs ? If so from which version ?

I suspect that the column format is not yet changed to string. So when the migration script change the value 8 to 8h0m0s it doesn't work. But I am not a expert in xorm and can't setup a similar env to replicate.
The more qualified people will maybe need those information for debug. ^^ 

For reference this is trigger by : go-gitea/gitea#1407

**Comment by strk** [comment]
@lunny can you look at this ?

**Comment by Lourens-Rich** [comment]
@sapk Yes, i migrated from Gogs, but idk which version have been used in last time. (0.9.~ It`s time when public start thinking about the new fork. I immediately went to Gitea master branch.)

**Comment by lunny** [comment]
@Lourens-Rich I think this should be caused by #1407 sent by @cez81 . Maybe you can change your custom/conf/app.ini about the mirror sync time interval to a time duration format to resolve this.

**Comment by cez81** [comment]
I think @sapk is on the right track. The column type might be the problem. In Sqlite which I use the column type is still integer after my change. Don't know if duration is handled differently in Mysql by xorm. The internal Go representation of duration is int64.

**Comment by lunny** [comment]
@cez81 you are right. XORM will not convert `int` to `bigint` automatically.

**Comment by bkcsoft** [comment]
This MySQL query should do the trick until there's a proper fix:
```sql
ALTER TABLE mirror MODIFY interval BIGINT;
```

**Comment by Lourens-Rich** [comment]
```
ALTER TABLE mirror MODIFY `interval` BIGINT;
```
on my database don`t fix the issue.

**Comment by cez81** [comment]
Can you confirm the current column type please?

**Comment by Lourens-Rich** [comment]
```
mysql> SELECT distinct DATA_TYPE FROM INFORMATION_SCHEMA.columns 
    -> WHERE table_schema = 'gogs' AND column_name like '%';
+-----------+
| DATA_TYPE |
+-----------+
| bigint    |
| int       |
| varchar   |
| tinyint   |
| text      |
| char      |
| smallint  |
+-----------+
7 rows in set (0.06 sec)
```
I don`t know how correctly take this information.

**Comment by cez81** [comment]
@Lourens-Rich 

`
SELECT column_name, data_type FROM INFORMATION_SCHEMA.columns WHERE table_schema = 'gogs' AND table_name = 'mirror'
`

**Comment by Lourens-Rich** [comment]
```
mysql> SELECT column_name, data_type FROM INFORMATION_SCHEMA.columns WHERE table_schema = 'gogs' AND table_name = 'mirror'
    -> ;
+------------------+-----------+
| column_name      | data_type |
+------------------+-----------+
| id               | bigint    |
| repo_id          | bigint    |
| interval         | bigint    |
| enable_prune     | tinyint   |
| updated_unix     | bigint    |
| next_update_unix | bigint    |
+------------------+-----------+
6 rows in set (0.01 sec)
```

**Comment by cez81** [comment]
Ok, that looks fine to me. And you still get this error message?

> 2017/04/16 00:16:59 [...itea/routers/init.go:54 GlobalInit()] [E] Failed to initialize ORM engine: migrate: do migrate: update mirror interval failed: Error 1264: Out of range value for column 'interval' at row 1

**Comment by Lourens-Rich** [comment]
```
./gitea web
2017/04/21 00:02:44 [T] Custom path: /home/git/go/src/code.gitea.io/gitea/custom
2017/04/21 00:02:44 [T] Log path: /home/git/go/src/github.com/gogits/gogs/log
{0 0} &{version migrations.Version [id version] map[id:[0xc420214540] version:[0xc4202146c0]] [0xc420214540 0xc4202146c0] map[] [id] id map[]    <nil>  } <nil>
```

gitea.log:
```
2017/04/20 23:10:35 [I] Log Mode: File(Info)
2017/04/20 23:10:35 [I] XORM Log Mode: File(Info)
2017/04/20 23:10:35 [I] Cache Service Enabled
2017/04/20 23:10:35 [I] Session Service Enabled
2017/04/20 23:10:35 [I] Migration: add field for repo size
2017/04/20 23:10:35 [I] This migration could take up to minutes, please be patient.
```

xorg.log: https://paste.fedoraproject.org/paste/T3Gn~mqLOj~1xI-pzRE5X15M1UNdIGYhyRLivL9gydE=xorm.log

**Comment by Lourens-Rich** [comment]
Stop. It was my fault, sorry.

**Comment by bkcsoft** [comment]
@Lourens-Rich Is this still an issue? Otherwise please close this issue :)

**Comment by cez81** [comment]
It's still an issue, need to merge https://github.com/go-gitea/gitea/pull/1504.

**Comment by lunny** [comment]
resolved by #1504
